types/react-redux/index.d.ts

```diff
@@ -15,12 +15,12 @@ type Store<S> = Redux.Store<S>;
 type Dispatch<S> = Redux.Dispatch<S>;
 type ActionCreator<A> = Redux.ActionCreator<A>;
 
-export interface DispatchProp<T> {
-  dispatch: Dispatch<T>
+export interface DispatchProp<S> {
+  dispatch: Dispatch<S>;
 }
 
-interface ComponentDecorator<TMergedProps> {
-    <TOwnProps>(component: Component<(TOwnProps & TMergedProps) | TOwnProps>): ComponentClass<TOwnProps>;
+interface ComponentDecorator<TMergedProps, TOwnProps> {
+    <T extends TOwnProps>(component: Component<T & TMergedProps>): ComponentClass<T>;
```

Comment:
ah interesting - I'll try that out tomorrow and see if both tests pass. Agree that it would be good to support both options. In the mean time, I'll also try the `connect()<OwnProps>(Component)`, happy for that to be the interim solution for my codebase while we try the other ideas. Appreciate the help! :)